interview-questions.md

```diff
@@ -8,42 +8,49 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is object-oriented programming? How is it different than functional programming?
 
-Your answer:
+Your answer: Object-oriented programming is used in response of accessing part of large amounts of data. OOP lets you take small snippets of the code base without affecting the whole entire part. Functional programming is only programming the parts that make it work.
 
-Researched answer:
+Researched answer: Object-oriented programming, OOP, is a way to create containers of data that can be changed and manipulated without affecting a large program. Functional programming is using functions to create clean and maintainable software.
 
 2. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: An integer is any whole number. A float is a number with a decimal value.
+For example, an integer can be the number 2 while a float can be 2.0.
 
-Researched answer:
+Researched answer: Integers in Ruby are any whole number that can be positive, negative, or zero numbers that do not have a fractional part. Floats are non-whole numbers that contain a fractional part or a decimal.
 
 3. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: An implicit return is the even though there is no return statement, Ruby will return the last line of the code block.
+Example: 
+```Ruby
+def num number
+    number
+end
+```
 
-Researched answer:
+Researched answer: An implicit return means that if there is no return statement, the value of the last line of a method is automatically returned.
```

Comment:
Correct answer, but what more can you tell me? What does that do compared to other approaches? How does that work?